bundles/org.openhab.binding.fineoffsetweatherstation/src/main/java/org/openhab/binding/fineoffsetweatherstation/internal/domain/MeasureType.java

```diff
@@ -0,0 +1,156 @@
+import static org.openhab.core.library.unit.Units.PERCENT;
+
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.function.BiFunction;
+
+import javax.measure.Unit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.fineoffsetweatherstation.internal.Utils;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.State;
+
+/**
+ * Represents the measured type with conversion form the sensors' bytes to the openHAB state.
+ *
+ * @author Andreas Berger - Initial contribution
+ */
+@NonNullByDefault
+public enum MeasureType {
+
+    TEMPERATURE(CELSIUS, 2, CHANNEL_TEMPERATURE, (data, offset) -> toInt16(data, offset) / 10.),
+
+    PERCENTAGE(PERCENT, 1, CHANNEL_HUMIDITY, (data, offset) -> toUInt8(data[offset])),
+
+    PRESSURE(HECTO(PASCAL), 2, CHANNEL_PRESSURE, Utils::toUInt16),
+
+    DEGREE(DEGREE_ANGLE, 2, null, Utils::toUInt16),
+
+    SPEED(METRE_PER_SECOND, 2, CHANNEL_WIND_SPEED, (data, offset) -> toUInt16(data, offset) / 10.),
+
+    HEIGHT(MILLI(METRE), 2, CHANNEL_RAIN, (data, offset) -> toUInt16(data, offset) / 10.),
+
+    HEIGHT_BIG(MILLI(METRE), 4, CHANNEL_RAIN, (data, offset) -> toUInt32(data, offset) / 10.),
+
+    HEIGHT_PER_HOUR(MILLIMETRE_PER_HOUR, 2, CHANNEL_RAIN_RATE, (data, offset) -> toUInt16(data, offset) / 10.),
+
+    LUX(Units.LUX, 4, CHANNEL_ILLUMINATION, (data, offset) -> toUInt32(data, offset) / 10.),
+
+    PM25(MICROGRAM_PER_CUBICMETRE, 2, CHANNEL_PM25, (data, offset) -> toUInt16(data, offset) / 10.),
+
+    PM10(MICROGRAM_PER_CUBICMETRE, 2, CHANNEL_PM10, (data, offset) -> toUInt16(data, offset) / 10.),
+
+    CO2(PARTS_PER_MILLION, 2, CHANNEL_CO2, Utils::toUInt16),
+
+    WATER_LEAK_DETECTION(1, CHANNEL_WATER_LEAK_DETECTION,
+            (data, offset) -> toUInt8(data[offset]) != 0 ? OnOffType.ON : OnOffType.OFF),
+
+    LIGHTNING_DISTANCE(KILO(METRE), 1, CHANNEL_LIGHTNING_DISTANCE, (data, offset) -> toUInt8(data[offset])),
+
+    LIGHTNING_COUNTER(4, CHANNEL_LIGHTNING_COUNTER, (data, offset) -> new DecimalType(toUInt32(data, offset))),
+
+    LIGHTNING_TIME(4, CHANNEL_LIGHTNING_TIME,
+            (data, offset) -> new DateTimeType(
+                    ZonedDateTime.ofInstant(Instant.ofEpochSecond(toUInt32(data, offset)), ZoneOffset.UTC))),
+
+    MICROWATT_PER_SQUARE_CENTIMETRE(Units.MICROWATT_PER_SQUARE_CENTIMETRE, 2, CHANNEL_UV_RADIATION, Utils::toUInt16),
+
+    BYTE(1, null, (data, offset) -> new DecimalType(toUInt8(data[offset]))),
+
+    DATE_TIME2(6, null, (data, offset) -> new DateTimeType(
+            ZonedDateTime.ofInstant(Instant.ofEpochSecond(toUInt32(data, offset)), ZoneOffset.UTC)));
```

Comment:
Please use `TimeZoneProvider` for creating DateTimeType in time-zone configured in openHAB.